### ISSUE TITLE
Nil pointer dereference during install when using a yaml config without proxyconfig

### ISSUE THREAD

**Issue by mbonnafon** [pydio/cells#256]
When running Pydio install with a YAML configuration file without the `proxyconfig` key it leads to a nil pointer dereference.
This happens regardless if the path is set with `CELLS_INSTALL_YAML` (docker case) or with the --yaml flag (binary) and even if I use a binary (2.1.3) or with the latest docker image pydio/cells from the docker hub.

You can reproduce it with an `install.yml` file without the `proxyconfig` key.

ProxyConfig seems optional https://github.com/pydio/cells/blob/master/common/proto/install/install.pb.go#L79 but if it's not provided `ProxyConfig` pointer is nil.
So, during the call for [applyProxyConfig](https://github.com/pydio/cells/blob/master/cmd/config-proxy.go#L96:6) since there is no check if the value is nil, it leads to a nil pointer error.

Should I consider that the `proxyconfig` attribute is not optional?

Stack trace: 
> cells_1  | ## Performing Installation
cells_1  | panic: runtime error: invalid memory address or nil pointer dereference
cells_1  | [signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x2a4a51c]
cells_1  |
cells_1  | goroutine 1 [running]:
cells_1  | github.com/pydio/cells/cmd.applyProxyConfig(0x0, 0x7f949c0, 0x4000e74a30)
cells_1  | 	/ext-go/1/src/github.com/pydio/cells/cmd/config-proxy.go:110 +0x17c
cells_1  | github.com/pydio/cells/cmd.installFromConf(0x1f, 0x0, 0x0)
cells_1  | 	/ext-go/1/src/github.com/pydio/cells/cmd/install-ni.go:131 +0x74
cells_1  | github.com/pydio/cells/cmd.nonInteractiveInstall(0x9b26aa0, 0x40009c0040, 0x2, 0x4, 0x0, 0x0, 0x6b)
cells_1  | 	/ext-go/1/src/github.com/pydio/cells/cmd/install-ni.go:42 +0x30
cells_1  | github.com/pydio/cells/cmd.glob..func33(0x9b26aa0, 0x40009c0040, 0x2, 0x4)
cells_1  | 	/ext-go/1/src/github.com/pydio/cells/cmd/install.go:162 +0x318
cells_1  | github.com/pydio/cells/vendor/github.com/spf13/cobra.(*Command).execute(0x9b26aa0, 0x40000dde40, 0x4, 0x4, 0x9b26aa0, 0x40000dde40)
cells_1  | 	/ext-go/1/src/github.com/pydio/cells/vendor/github.com/spf13/cobra/command.go:760 +0x1cc
cells_1  | github.com/pydio/cells/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x9b27b40, 0x40003f2bd8, 0x1, 0x1)
cells_1  | 	/ext-go/1/src/github.com/pydio/cells/vendor/github.com/spf13/cobra/command.go:846 +0x224
cells_1  | github.com/pydio/cells/vendor/github.com/spf13/cobra.(*Command).Execute(...)
cells_1  | 	/ext-go/1/src/github.com/pydio/cells/vendor/github.com/spf13/cobra/command.go:794
cells_1  | github.com/pydio/cells/cmd.Execute()
cells_1  | 	/ext-go/1/src/github.com/pydio/cells/cmd/root.go:201 +0x124
cells_1  | main.main()
cells_1  | 	/ext-go/1/src/github.com/pydio/cells/main.go:119 +0x88

**Comment by bsinou** [comment]
Hello, 

TL;DR
>Should I consider that the proxy-config attribute is not optional?
In your case, yes.

A few hints:
- this non-interactive yaml install is still a tech preview and will be broken in the next major version when we introduce support for multiple public URLs.
- the proto you are pointing to is both used by the web installer and by the non-interactive install => thus the optional: the proxy config is already defined when we reach this point with the web installer.
- the proxy config is also compulsory in your case to define the configuration of the main gateway of the app 

Thus said, thanks for reporting,  we will double check when preparing the next major version to make this more reliable / documented.

**Comment by mbonnafon** [comment]
Hi, thank you for this quick answer and theses hints!
